File: Mobiles/flutter/lib/utils/faro_utils.dart
/// Utility functions for Grafana Faro configuration
library;

/// Extracts the API key token from a Grafana Faro collector URL.
///
/// The token is expected to be the path segment immediately after `/collect/`.
/// Format: https://domain/collect/TOKEN
///
/// Returns an empty string if:
/// - The URL is null or empty
/// - The URL doesn't contain `/collect/`
/// - The token segment is empty
/// - URL parsing fails
String extractTokenFromCollectorUrl(String? collectorUrl) {
  if (collectorUrl == null || collectorUrl.isEmpty) {
    return '';
  }

  try {
    final uri = Uri.parse(collectorUrl);
    final pathSegments = uri.pathSegments;
    final collectIndex = pathSegments.indexOf('collect');
    if (collectIndex != -1 && collectIndex + 1 < pathSegments.length) {
      final token = pathSegments[collectIndex + 1];
      if (token.isNotEmpty) {
        return token;
      }
    }
  } catch (e) {
    return '';
  }

  return '';
}

File: Mobiles/react-native/src/core/utils/faroUtils.ts
/**
 * Extracts the API key token from a Grafana Faro collector URL.
 * Format: https://domain/collect/TOKEN
 */
export function extractTokenFromCollectorUrl(collectorUrl: string | undefined): string {
  if (!collectorUrl || collectorUrl.trim().length === 0) {
    return '';
  }

  try {
    const url = new URL(collectorUrl);
    const segments = url.pathname.split('/').filter(Boolean);
    const collectIndex = segments.indexOf('collect');
    if (collectIndex !== -1 && collectIndex + 1 < segments.length) {
      const token = segments[collectIndex + 1];
      if (token && token.length > 0) {
        return token;
      }
    }
  } catch {
    return '';
  }

  return '';
}
